mail_notify_bounce/models/mail_thread.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Lorenzo Battistini - Agile Business Group
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+import logging
+from odoo import models, api, tools
+
+_logger = logging.getLogger(__name__)
+
+
+class MailThread(models.AbstractModel):
+    _inherit = 'mail.thread'
+
+    @api.model
+    def message_route(
+        self, message, message_dict, model=None, thread_id=None,
+        custom_values=None
+    ):
+        if self.env.context.get('fetchmail_server_id'):
```

Comment:
move all of this to its own function, this allows other modules to intercept this if they need to